gstd/src/common/handlers.rs

```diff
@@ -25,43 +25,45 @@
 //! debug and non-debug mode, for programs built in `wasm32` architecture.
 //! For `debug` mode it provides more extensive logging.
 
-#[cfg(target_arch = "wasm32")]
-use {crate::ext, core::alloc::Layout, core::panic::PanicInfo};
-
-#[cfg(target_arch = "wasm32")]
+#[cfg(feature = "oom-handler")]
 #[alloc_error_handler]
-pub fn oom(_: Layout) -> ! {
-    ext::oom_panic()
+pub fn oom(_: core::alloc::Layout) -> ! {
+    crate::ext::oom_panic()
 }
 
-#[cfg(not(any(feature = "debug", debug_assertions)))]
-#[cfg(target_arch = "wasm32")]
-#[panic_handler]
-pub fn panic(_: &PanicInfo) -> ! {
-    ext::panic("no info")
-}
+#[cfg(feature = "panic-handler")]
+mod panic_handler {
+    use crate::ext;
+    use core::panic::PanicInfo;
 
-#[cfg(any(feature = "debug", debug_assertions))]
-#[cfg(target_arch = "wasm32")]
-#[panic_handler]
-pub fn panic(panic_info: &PanicInfo) -> ! {
-    use crate::prelude::format;
+    #[cfg(not(all(feature = "panic-messages", feature = "debug")))]
+    #[panic_handler]
+    pub fn panic(_: &PanicInfo) -> ! {
+        ext::panic("no info")
+    }
 
-    let msg = match (panic_info.message(), panic_info.location()) {
-        (Some(msg), Some(loc)) => format!(
-            "'{:?}', {}:{}:{}",
-            msg,
-            loc.file(),
-            loc.line(),
-            loc.column()
-        ),
-        (Some(msg), None) => format!("'{msg:?}'"),
-        (None, Some(loc)) => {
-            format!("{}:{}:{}", loc.file(), loc.line(), loc.column())
-        }
-        _ => ext::panic("no info"),
-    };
+    #[cfg(all(feature = "panic-messages", feature = "debug"))]
+    #[panic_handler]
+    pub fn panic(panic_info: &PanicInfo) -> ! {
+        use crate::prelude::format;
+        let msg = match (panic_info.message(), panic_info.location()) {
```

Comment:
```suggestion
        #[cfg(all(feature = "panic-messages", feature = "debug"))]
        let message = panic_info.message();
        #[cfg(not(all(feature = "panic-messages", feature = "debug")))]
        let message = None;

        let msg = match (message, panic_info.location()) {
```

Reply:
Makes sense

Reply:
Done in 5d9baa86f.

Reply:
Had to be partly undone in f12f67055 since the `debug` and non-`debug` implementations still need to be separated.